### DISCUSSION TITLE
what's the different with the se_a and se_r and se_ar in the json file？

### DISCUSSION THREAD

**Discussion by GitWhilebear** [deepmodeling/deepmd-kit#126] [Q&A] [answered]
hi 
I'm a little confused. what's the different with the param se_a and se_r and se_ar in the json file. How do I choose the type when I train a model.
thank a lot.

**Comment by jameswind** [comment, marked as answer]
This is something not well documented. 
For the DeepPot-SE model, in the original paper (http://papers.nips.cc/paper/7696-end-to-end-symmetry-preserving-inter-atomic-potential-energy-model-for-finite-and-extended-systems) we define NN-parameterized descriptors D=G_1^TRR^TG_2, see Eqn. 11, where G are NN parameterized embedding functions and R are generalized coordinate f(r)*(r,x,y,z). In the code, we use se_a (short for selected angular input) to denote the largest number of neighbors selected in r_cut and we will use the coordinates of these neighbors to define the G, R, etc., introduced here.

Later we realized that for neighbors that are far from an adopted atom, only the radial part in R are essential. So we introduce some additional descriptors with the only difference from D being that the (x,y,z) components in R are removed. In the code, we use se_r  (short for selected radial input) to denote the largest number of neighbors that contribute the radial descriptor.

Notice that the r_cut for angular and radial descriptors can be different, and that for the latter case is usually larger. The example corresponding to s_ar simply combines the two types of descriptors to construct the DP model.

**Comment by GitWhilebear** [comment]
I get it, thank you very much.
but I have an other question, in the paper, there is a description of local embeding network G:

> Next, we define the local embedding network ...... ,a neural network mapping from a single value s(rji), through multiple hidden layers, to M1 outputs.

Is [s(r1i)... s(rji) ...]T the input of local embding network G? Why [s(r1i)... s(rji) ...]T (part of Ri) is the input of local embding network G instead of [s(rji) x y z] (all of Ri)？ In my understanding, when the  descriptor use the architecture proposed in ref.20 to preserve  permutation invariance, it need use the  [s(rji) x y z] (all of Ri) as a set data as the input of local embding network G, am I right?

**Comment by jameswind** [comment]
If one uses as inputs to G the full information including x y z, the rotational symmetry will not be exactly preserved.
